rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java

```diff
@@ -108,6 +110,37 @@ public ResponseEntity<EntityModel<Vendor>> getVendor(
         return new ResponseEntity<>(halResource, HttpStatus.OK);
     }
 
+    @Operation(
+            summary = "Delete a vendor.",
+            description = "Delete vendor by id.",
+            tags = {"Vendor"}
+    )
+    @RequestMapping(value = VENDORS_URL + "/{id}", method = RequestMethod.DELETE)
+    public ResponseEntity<?> deleteVendor(
+            @Parameter(description = "The id of the vendor to be deleted.")
+            @PathVariable("id") String id
+    ) {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        boolean idFound = false;
+        List<Vendor> vendors = vendorService.getVendors();
+        for (Vendor vendor : vendors) {
+            if (vendor.getId().equals(id)) {
+                idFound = true;
+                break;
+            }
+        }
```

Comment:
Do we not have something like `getVendorById()` ? That would be much more efficient.

Reply:
made the changes.